### ISSUE TITLE
Телеграмм-канал с событиями

### ISSUE THREAD

**Issue by dmitriym09** [web-standards-ru/calendar#289]
Предлагаю сделать телеграмм-канал с автопостингом всех событий, добавляемых в календарь (по хуку например). Как считаете, имеет ли это смысл и нужно ли это кому?

**Comment by pepelsbey** [comment]
А давайте :) Я создал канал [webstandards_events](https://t.me/webstandards_events), давайте будем постить туда. Нужно только разобраться в формате.

**Comment by dmitriym09** [comment]
На чем бэк же сделаю - не важно? Планирую на python - надеюсь не вызовет несогласия. Распространять планирую в docker.  Сейчас я сделал у себя приватный репозиторий, в нем напишу тестовый бот в тестовый канал (имя отдельно сообщу) и запущу его на своих ресурсах. Тестовый будет сканировать наличие новых событий (как устаканется, можно навесить хуки). В сообщении думаю посылать название, сроки, ссылку и gps-координаты на карте места события. Сразу предложение - можно опционально указывать точное местоположение - над форматом подумаю.

Сроки - надеюсь к концу следующей недели постараюсь выкатить тест.

**Comment by pepelsbey** [comment]
А нужен бэк? Мы можем наверное на Тревисе запускать сборку, как мы уже сейчас собираем календарь, и публиковать событие. Если оно там так может сработать.

**Comment by pepelsbey** [comment]
Я бы конечно предпочёл всё на JS :) У нас больше шансов, что скрипт будет жить и его можно будет легко починить.

**Comment by pepelsbey** [comment]
Мы уже деплоим календарь на наши хосты с помощью Тревиса. То есть все (зашифрованные) ключи уже розданы.

Посмотрите, как оно в этом репозитории устроено. Грубо говоря, если вы сможете локально написать npm-скрипт, который будет публиковать что-то в Телеграм, значит это сможет сделать и Тревис удалённо, после комита в мастер.

**Comment by dmitriym09** [comment]
Ок, понял. В принципе бот - это скрипт, который и публикую что-то в телеграмм, но весит в фоне. Соответственно,  я реализую npm-модуль, который отправляет событие в канал.  А интеграцию с трэвисом думаю поможете. Вернее это вы и сможете только сделать, как имеющие доступ ))

**Comment by dmitriym09** [comment]
Ну нужно иметь в виду ркн. В настойках скрипта можно будет передать прокси.

**Comment by pepelsbey** [comment]
Скрипт будет запускаться вне юрисдикции РКН, на серверах Тревиса.

**Comment by dmitriym09** [comment]
Ок. Как я понимаю удобнее всего будет мне выгрузить модуль на https://www.npmjs.com/, откуда вы его возьмете и внедрите его?

**Comment by pepelsbey** [comment]
Да, если брать модуль, то из npm. И писать всё самостоятельно.

**Comment by dmitriym09** [comment]
"И писать всё самостоятельно." - что имеется в виду? Нельзя использовать сторонние модули?

**Comment by dmitriym09** [comment]
Или в смысле что сам модуль должен быть мной написан, а не попытка его свалить на вас?

**Comment by pepelsbey** [comment]
Опечатка, поправил сообщение.

**Comment by dmitriym09** [comment]
Ок, понятно. Набросал уже тестовый модуль. На следующей неделе надеюсь, что доведу до состояния, когда можно опубликовать. И тогда уже обсудим

**Comment by dmitriym09** [comment]
https://github.com/dmitriym09/web-standards-ru-calendar-telegram

**Comment by dmitriym09** [comment]
Вот сделал и опубликовал тестовую версию модуля. Пока что в реестр не выкладывал. На днях доведу до ума, документирую и подчищу код. Модуль сейчас рабочий - думаю понятен подход. Жду фидбека по удобству и возможности внедрения.

**Comment by dmitriym09** [comment]
Добавил в реестр `npm i -S web-standards-ru-calendar-telegram`, методы документировал. Жду фидбека

**Comment by pepelsbey** [comment]
Ох, не успел написать, что публиковать в npm не нужно, это же очень узкий скрипт, который проще положить рядом, как уже лежит скрипт для сборки календаря в файл.

**Comment by dmitriym09** [comment]
Могу запустить автопостинг на своих ресурсах, если дадите доступ моему боту. Потом как будет время настроите Трэвис и отключим мой бот, а пока он запустит все существующие события и будет следить за новыми.

**Comment by pepelsbey** [comment]
Я тут подумал, что наверное имеет смысл перетащить все наши пакеты в npm-организацию. Так что позвал туда. Давай опубликуем calendar-bot там?

**Comment by dmitriym09** [comment]
Хочу перенести https://github.com/dmitriym09/calendar-bot в https://github.com/web-standards-ru/calendar-bot - не хватает прав

**Comment by pepelsbey** [comment]
@dmitriym09 добавилось два события, но опубликовались дубликаты одного:

<img width="229" alt="image" src="https://user-images.githubusercontent.com/105274/59162039-14abcc00-8af3-11e9-9eaa-a539a189c375.png">

<img width="381" alt="image" src="https://user-images.githubusercontent.com/105274/59162045-22f9e800-8af3-11e9-9129-ca4a2db1c7ab.png">

**Comment by dmitriym09** [comment]
@pepelsbey проблема оказалась в том, что было быстро добавлено 2 коммита, и оба travis контейнера получили последнюю версия репозитория. Не продумал данный кейс. Думаю пути решения - пока в голову приходит только читать канал в боте и искать новые события каждый раз. Может еще кто что предложит? Пока просьба, между коммитами выдерживать паузу, чтобы успевал предыдущий отослаться.

**Comment by dmitriym09** [comment]
@pepelsbey проблема решается отключением Auto Cancellation в travis (я уже отключил). Это заставит отрабатывать контейнер для каждого коммита не прерывая его даже если появились новые в процессе работы. Тесты показали, что работает. Будем посмотреть.

А для бота сделать читалку канала кстати нельзя - это нужно писать приложение, что сложнее.

**Comment by m0rg0t** [comment]
Функционирует ли механизм автопостинга в канал в настоящее время?

Так как события в канале последние от 29 апреля, но новые события в том же мае добавлялись.

![image](https://user-images.githubusercontent.com/634236/82316522-95e33280-99d5-11ea-9727-4efeb6ae4ac1.png)

P.S. Также https://web-standards.ru/calendar.json в настоящий момент устаревший, без свежедобавленных событий.

**Comment by baitun** [comment]
Хм, действительно... 
@pepelsbey, выглядит, как будто, после этого коммита https://github.com/web-standards-ru/calendar/commit/07d6a3918aaf2c7aefc2df8730558b78b774cd68, CI перестал работать

**Comment by pepelsbey** [comment]
Да, мы переехали на новый сервер и пока публикация не работает. Буду чинить вечером.

**Comment by pepelsbey** [comment]
Автодеплой снова работает, теперь на Actions, без Travis.

**Comment by m0rg0t** [comment]
Вечер добрый!

Телеграмм канал с событиями пока до сих пор с апрельскими событиями, так что с публикацией потенциально проблема все ещё осталась

![0969AF40-608C-4A69-9EDF-74C862A30C33](https://user-images.githubusercontent.com/634236/94374377-8f4b9100-0114-11eb-831f-0b16c9eba4c6.png)

**Comment by Manyaka** [comment]
Может уже появилось время поправить то, что последнее событие в канале от 24 апреля 2020?

**Comment by pepelsbey** [comment]
@Manyaka нужен активист, который возьмёт в руки решение проблем и разработку — попытки были, но до конца так и не дошли (

**Comment by dmitriym09** [comment]
> @Manyaka нужен активист, который возьмёт в руки решение проблем и разработку — попытки были, но до конца так и не дошли (

Весь код написан и тесты проходит. Проблема возникла - никто не захотел сделать код-ревью. На этом все и зависло.

**Comment by pepelsbey** [comment]
> Проблема возникла - никто не захотел сделать код-ревью

Никто не смог сделать код ревью: очень много кода, непонятно что именно он делает. Предлагаю вернуться [в пулреквест](https://github.com/web-standards-ru/calendar-bot/pull/15) и продолжить там.
